### PR TITLE
[BugFix] Nested keys in transforms

### PR DESCRIPTION
**Pull request by matteobettini**

### CODE REVIEW

File: torchrl/envs/transforms/transforms.py [hunk 1]
```diff
@@ -2489,8 +2489,10 @@ def __init__(
                 )
         else:
             in_keys = sorted(in_keys, key=_sort_keys)
-        if type(out_key) != str:
-            raise Exception("CatTensors requires out_key to be of type string")
+        if not isinstance(out_key, str) and not (
+            isinstance(out_key, tuple) and set(map(type, out_key)) == {str}
+        ):
```
Comment (vmoens) [inline torchrl/envs/transforms/transforms.py:2494]: See https://github.com/pytorch-labs/tensordict/pull/479/files#r1254257836

I wonder if this is needed

**Review by vmoens — APPROVED** [review]
LGTM very thorough!

One small comment

File: torchrl/envs/transforms/transforms.py (hunk 1 above)
Comment (matteobettini) [reply torchrl/envs/transforms/transforms.py:2494]: should we just chek it is not a list?

**Comment by vmoens** [comment]
examples CI seems to be broken, not sure it's related to this PR (maybe Double2Float?)

**Comment by matteobettini** [comment]
oh thanks let me have a look

**Pull request merged**